ocaml/jane/doc/extensions/local/reference.md

```diff
@@ -502,46 +499,38 @@ let f () =
 In this code, the counter will *not* be allocated locally. The reason is the
 `Counter.make` function: the allocation of `ref 0` escapes the region of
 `Counter.make`, and the compiler will therefore not allow it to be locally
-allocated. This remains the case no matter how many local_ annotations we write
-inside `f`: the issue is the definition of `make`, not its uses.
+allocated. This remains the case no matter how many `local_` annotations we
+write inside `f`: the issue is the definition of `make`, not its uses.
 
-To allow the counter to be locally allocated, we need to specify that
-`Counter.make` may return local allocations. This can be done by wrapping the
-entire body of `make` with the `local_` keyword:
+To allow the counter to be locally allocated, we need to make `Counter.make`
+allocate and return in caller's region. This can be done by `exclave_`:
 
 ```ocaml
-let make () = local_
+let make () = exclave_
   ref 0
 ```
 
-The `local_` keyword around a function body like this specifies not only that
-the allocation of the `ref` should be local, but more importantly that the
-function `make` *should not have its own region*.
-
-Instead, local allocations during `make` are considered part of `f`s region,
-and will only be cleaned up when that region ends. Local allocations are
-allocated as always in the nearest enclosing region. However if the current
-function is a local-returning function, then the nearest enclosing region will
-be the caller's (or that of the caller's caller, etc., if the caller is also
-local-returning).
+The keyword `exclave_` terminates the current region and executes the subsequent
+code in the outer region. Therefore, `ref 0` is executed in `f`'s region, which
+allows its local allocation. The allocation will only be cleaned up when the
```

Comment:
Note to self: when we merge this PR, we should make a follow-on PR that uses the term "stack allocation" consistently instead of "local allocation".

Reply:
Yes I think [#2540](https://github.com/ocaml-flambda/flambda-backend/pull/2540#discussion_r1653222006) is very related. My current plan is that that PR will be merged without document and notifying the user. We will (in a few weeks) do a complete overhaul of this document, talk about "stack allocation" and introduce the `stack_` syntax.